sale_order_weight/models/sale_order_line.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2021 Manuel Calero Solís (https://xtendoo.es)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    total_ordered_weight = fields.Float(
+        compute="_compute_total_ordered_weight",
+        string="Ordered Weight",
+        store=True,
+    )
+
+    @api.depends(
+        "product_id.weight", "product_id.uom_id", "product_uom", "product_uom_qty"
+    )
+    def _compute_total_ordered_weight(self):
+        for line in self.filtered(lambda x: x.product_id):
```

Comment:
pre-filtering in such cases causes to loop twice on the same records. As you have a loop, just filter inside it:
```
for line in self:
    if not line.product_id:
        continue
[...]
```
BTW... can you have a line w/o product?

Reply:
> pre-filtering in such cases causes to loop twice on the same records. As you have a loop, just filter inside it:

Even though I've been working with odoo for a few years now, I still find it hard to understand certain optimisations for using ORM. I need to study again the odoo experience talks! 

> BTW... can you have a line w/o product?

Yes, in sale.order.line model, you have 3 "type" of lines. "product", "section", and "note". section and note doesn't have a product_id defined.

Reply:
@pedrobaeza : thanks for your extra explanation.

@simahawk Are you agree with @pedrobaeza remarks ? what is the definive syntax to use ? 

Thanks !

Reply:
I disagree, about the code being uglier, for sure :smile: 
Yet, this is not a blocker.